### DISCUSSION TITLE
Reusable workflow not detecting pull_request closed event

### DISCUSSION THREAD

**Discussion by fiddlermikey** [community/community#70686] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

How do I debug/troubleshoot why a trigger is not detecting an event?

I have a reusable workflow (github-release.yml) being called by a starter workflow (starter-workflow.yml). The github-release.yml workflow returns values (as outputs) to the caller workflow (starter-workflow.yml). 

The issue that that the reusable workflow (github-release.yml) only runs when the starter-workflow.yml triggers github-release.yml on push, and pull_request synchronize and open events. The reusable workflow is not called when pull_request is closed. The commit from the closed pull_request event still activates the push trigger, but does not seem to recognize the closed pull_request.

How do I troubleshoot this?

Pertinent sections of workflows are as follows:

### starter-workflow.yml

~~~
name: Starter Workflow
on: [workflow_dispatch, push, pull_request]

jobs:
  call-create-github-release-workflow:
    uses: MyOrg/actions/.github/workflows/github-release.yml@support-hotfixes
~~~

### github-release.yml
This one only runs for pull_request: opened and synchrnonized but not for closed events
~~~
name: Create GitHub release version and location
on:
  workflow_call:
    outputs:
      release_version:
        description: The computed release version for the release (or pre-release) being built
        value: ${{ jobs.create-github-release.outputs.release_version }}
      release_url:
        description: The upload URL for build artifacts for this release
        value: ${{ jobs.create-github-release.outputs.release_url }}
  pull_request:
  push:
~~~

Here is a screenshot of the actions that ran, and the events that triggered them.
The commit from the closed pull_request event still activates the push trigger, but does not seem to recognize the closed pull_request.
Events:

- 443: Create PR
- 444: Commit to the pr incoming branch
- 445: Additional commit to incoming branch
- 446: Synchronize event created by 445
- Merge PR
- 447: push event runs, but not pull_request (closed)

![failed_pr_trigger.png](https://github.zendesk.com/attachments/token/5IUIqIfwjaUZCoDByJOaJw860/?name=failed_pr_trigger.png)

**Comment by fiddlermikey** [comment]
I created a couple of repositories with a caller and a reusable workflow illustrating this issue

Caller - https://github.com/fiddlermikey/pr-trigger-demo
Calls reusable - https://github.com/fiddlermikey/my-actions

**Comment by seyLu** [comment, marked as answer]
#### Hi @fiddlermikey,

![image](https://github.com/community/community/assets/98249191/6dc9100b-95e0-4657-abc3-ae08c40d327b)
According to the docs, if you do not specify activity type, it defaults to `opened` or `reopened`.

You can read more on [Github docs: Events that trigger workflows (`pull_request`)](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)

**Reply by fiddlermikey** [reply]
Thank you. I added the synchronize event and the build is triggering appropriately.
